src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -676,21 +674,23 @@ bool SotaUptaneClient::downloadImages(const std::vector<Uptane::Target> &targets
     // TODO: support downloading encrypted targets from director
     // TODO: check if the file is already there before downloading
     if (!uptane_fetcher->fetchVerifyTarget(*images_target)) {
-      *events_channel << std::make_shared<event::Error>("Error downloading targets.");
+      sendEvent(std::make_shared<event::Error>("Error downloading targets."));
       return false;
     }
   }
   if (!targets.empty()) {
     if (targets.size() == downloaded_targets.size()) {
-      *events_channel << std::make_shared<event::DownloadComplete>(downloaded_targets);
+      sendEvent(std::make_shared<event::DownloadComplete>(downloaded_targets));
       sendDownloadReport();
+      if (config.uptane.running_mode == RunningMode::kFull || config.uptane.running_mode == RunningMode::kOnce) {
+        uptaneInstall(downloaded_targets);
```

Comment:
That's still a bit annoying that all these operations link to each others internally depending on the running mode, which obfuscates the flow control.

Do you think it would be possible to have an `uptaneCycle()` which would subscribe to events internally to determine the next operation. `RunninMode::kOnce` would just call this once, `kFull` in a loop and the other single ops would just call other operations à la carte.

End-goal: no reading of `config.uptane.running_mode` in `sotauptaneclient.cc`.

Reply:
> That's still a bit annoying that all these operations link to each others internally depending on the running mode, which obfuscates the flow control.

Agreed, and I think your end-goal is reasonable. I will consider the `uptaneCycle` option or do something similar.